### PR TITLE
⬆️ JAVA-3298 Upgrade Contrast Java SDK

### PR DESCRIPTION
**Pull request by gilday**
The new Contrast SDK has Scan support built-in. Use SDK's new API for setting the User-Agent header.

The build will not pass until we have released the latest contrast-java-sdk (🔜 ). At that time, I will configure the build to run end-to-end tests with the Contrast staging environment.

### CODE REVIEW

File: src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java [hunk 1]
```diff
@@ -131,33 +131,35 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
     // get or create project
-    final Project project = findOrCreateProject(contrastScan);
+    final Project project = findOrCreateProject(projects);
+
+    // upload code artifact
+    getLog().info("Uploading " + file.getFileName() + " to Contrast Scan");
+    final CodeArtifact codeArtifact;
+    try {
+      codeArtifact = project.codeArtifacts().upload(file);
+    } catch (final IOException | HttpResponseException e) {
```
Comment (tylernhoward) [inline src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java:141]: Ahhhh beautiful multicatches

File: src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java [hunk 2]
```diff
@@ -137,19 +149,30 @@ void setOrganizationId(final String organizationId) {
...
           .build();
-    } catch (IllegalArgumentException e) {
+    } catch (final IllegalArgumentException e) {
       throw new MojoExecutionException(
           "\n\nWe couldn't connect to Contrast at this address [" + url + "]. The error is: ", e);
     }
   }
 
+  /**
+   * visible for testing
```
Comment (tylernhoward) [inline src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java:167]: you dont want to create a marker annotation?

File: src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java [hunk 3]
```diff
@@ -131,33 +131,35 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+      throw new MojoExecutionException("Failed to upload code artifact to Contrast Scan", e);
```
Comment (tylernhoward) [inline src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java:142]: `MojoFailureException` , no?

File: src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java [hunk 4]
```diff
@@ -325,6 +337,8 @@ private void waitForResults(final ScanOperation operation)
               : (duration.toMillis() / 1000) + " seconds";
       throw new MojoFailureException(
           "Failed to retrieve Contrast Scan results in " + durationString, e);
+    } finally {
```
Comment (tylernhoward) [inline src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java:340]: i always wish this implemented closeable, but i suppose it wouldn't due to `.shutdown` and `.shutdownNow` both being options to interact with the service. I suppose you could right your own `AutoClosable` that delegated to the `shutdown` method. Anyway, ignore my ramblings - its fine.

**Review by tylernhoward — APPROVED** [review]

File: src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java (hunk 2 above)
Comment (gilday) [reply src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java:167]: Not really. We never do anything with the `VisibleForTesting` annotation.

File: src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java (hunk 4 above)
Comment (gilday) [reply src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java:340]: Just be patient and wait for [structured concurrency](https://wiki.openjdk.java.net/display/loom/Structured+Concurrency) ;-)

File: src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java (hunk 3 above)
Comment (gilday) [reply src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java:142]: Yeah I'll update them all

**Review by tylernhoward — APPROVED** [review]

**Pull request merged**